diff --no-dereference -ruN a/src/animation/animation.c b/src/animation/animation.c
--- a/src/animation/animation.c	2026-07-28 15:15:18.738296889 +0300
+++ b/src/animation/animation.c	2026-07-28 15:15:18.808530008 +0300
@@ -81,6 +81,7 @@
  */
 
 #include <GL/glu.h>
+#include <stdlib.h>
 #include "animation-internal.h"
 
 
diff --no-dereference -ruN a/src/animation/curvedfold.c b/src/animation/curvedfold.c
--- a/src/animation/curvedfold.c	2026-07-28 15:15:18.738391298 +0300
+++ b/src/animation/curvedfold.c	2026-07-28 15:15:18.814502247 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Curved Fold  =========================
 
diff --no-dereference -ruN a/src/animation/dodge.c b/src/animation/dodge.c
--- a/src/animation/dodge.c	2026-07-28 15:15:18.738501278 +0300
+++ b/src/animation/dodge.c	2026-07-28 15:15:18.818639828 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Dodge  =========================
 
diff --no-dereference -ruN a/src/animation/dream.c b/src/animation/dream.c
--- a/src/animation/dream.c	2026-07-28 15:15:18.738424634 +0300
+++ b/src/animation/dream.c	2026-07-28 15:15:18.822808557 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 Bool
 fxDreamAnimInit (CompWindow * w)
diff --no-dereference -ruN a/src/animation/fade.c b/src/animation/fade.c
--- a/src/animation/fade.c	2026-07-28 15:15:18.738155801 +0300
+++ b/src/animation/fade.c	2026-07-28 15:15:18.827523925 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Fade  =========================
 
diff --no-dereference -ruN a/src/animation/focusfade.c b/src/animation/focusfade.c
--- a/src/animation/focusfade.c	2026-07-28 15:15:18.738136970 +0300
+++ b/src/animation/focusfade.c	2026-07-28 15:15:18.831862908 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Focus Fade  =========================
 
diff --no-dereference -ruN a/src/animation/glide.c b/src/animation/glide.c
--- a/src/animation/glide.c	2026-07-28 15:15:18.738114818 +0300
+++ b/src/animation/glide.c	2026-07-28 15:15:18.835696592 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Glide  =========================
 
diff --no-dereference -ruN a/src/animation/horizontalfold.c b/src/animation/horizontalfold.c
--- a/src/animation/horizontalfold.c	2026-07-28 15:15:18.738469125 +0300
+++ b/src/animation/horizontalfold.c	2026-07-28 15:15:18.839491568 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Horizontal Folds  =========================
 
diff --no-dereference -ruN a/src/animation/magiclamp.c b/src/animation/magiclamp.c
--- a/src/animation/magiclamp.c	2026-07-28 15:15:18.738444768 +0300
+++ b/src/animation/magiclamp.c	2026-07-28 15:15:18.843535096 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 void
 fxMagicLampInitGrid (CompWindow *w,
diff --no-dereference -ruN a/src/animation/options.c b/src/animation/options.c
--- a/src/animation/options.c	2026-07-28 15:15:18.738193704 +0300
+++ b/src/animation/options.c	2026-07-28 15:15:18.847346125 +0300
@@ -26,6 +26,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 extern ExtensionPluginInfo animExtensionPluginInfo;
 extern AnimBaseFunctions animBaseFunctions;
diff --no-dereference -ruN a/src/animation/rollup.c b/src/animation/rollup.c
--- a/src/animation/rollup.c	2026-07-28 15:15:18.738173982 +0300
+++ b/src/animation/rollup.c	2026-07-28 15:15:18.851125338 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Roll Up  =========================
 
diff --no-dereference -ruN a/src/animation/wave.c b/src/animation/wave.c
--- a/src/animation/wave.c	2026-07-28 15:15:18.738527077 +0300
+++ b/src/animation/wave.c	2026-07-28 15:15:18.855120754 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Wave  =========================
 
diff --no-dereference -ruN a/src/animation/zoomside.c b/src/animation/zoomside.c
--- a/src/animation/zoomside.c	2026-07-28 15:15:18.738236296 +0300
+++ b/src/animation/zoomside.c	2026-07-28 15:15:18.858814991 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Zoom and Sidekick  =========================
 
diff --no-dereference -ruN a/src/mag/mag.c b/src/mag/mag.c
--- a/src/mag/mag.c	2026-07-28 15:15:18.736543023 +0300
+++ b/src/mag/mag.c	2026-07-28 15:15:18.862830425 +0300
@@ -21,6 +21,7 @@
  */
 
 #include <math.h>
+#include <stdlib.h>
 #include <string.h>
 
 #include <compiz-core.h>
diff --no-dereference -ruN a/src/mousepoll/mousepoll.c b/src/mousepoll/mousepoll.c
--- a/src/mousepoll/mousepoll.c	2026-07-28 15:15:18.737002736 +0300
+++ b/src/mousepoll/mousepoll.c	2026-07-28 15:15:18.867250806 +0300
@@ -21,6 +21,7 @@
  */
 
 #include <compiz-core.h>
+#include <stdlib.h>
 
 #include "compiz-mousepoll.h"
 
diff --no-dereference -ruN a/src/opacify/opacify.c b/src/opacify/opacify.c
--- a/src/opacify/opacify.c	2026-07-28 15:15:18.735797543 +0300
+++ b/src/opacify/opacify.c	2026-07-28 15:15:18.871026902 +0300
@@ -22,6 +22,7 @@
  */
 
 #include <compiz-core.h>
+#include <stdlib.h>
 #include "opacify_options.h"
 
 #define GET_OPACIFY_DISPLAY(d)                            \
diff --no-dereference -ruN a/src/scaleaddon/scaleaddon.c b/src/scaleaddon/scaleaddon.c
--- a/src/scaleaddon/scaleaddon.c	2026-07-28 15:15:18.736006437 +0300
+++ b/src/scaleaddon/scaleaddon.c	2026-07-28 15:15:18.874686476 +0300
@@ -25,6 +25,7 @@
  */
 
 #include <math.h>
+#include <stdlib.h>
 #include <string.h>
 #include <X11/Xatom.h>
 
diff --no-dereference -ruN a/src/session/session.c b/src/session/session.c
--- a/src/session/session.c	2026-07-28 15:15:18.737698139 +0300
+++ b/src/session/session.c	2026-07-28 15:15:18.878602086 +0300
@@ -25,6 +25,7 @@
 
 #define _GNU_SOURCE
 #include <X11/Xatom.h>
+#include <stdlib.h>
 #include <X11/SM/SM.h>
 
 #include <compiz-core.h>
diff --no-dereference -ruN a/src/vpswitch/vpswitch.c b/src/vpswitch/vpswitch.c
--- a/src/vpswitch/vpswitch.c	2026-07-28 15:15:18.735274800 +0300
+++ b/src/vpswitch/vpswitch.c	2026-07-28 15:15:18.882827942 +0300
@@ -27,6 +27,7 @@
  */
 
 #include <compiz-core.h>
+#include <stdlib.h>
 #include <string.h>
 #include <X11/keysymdef.h>
 #include "vpswitch_options.h"
diff --no-dereference -ruN a/src/winrules/winrules.c b/src/winrules/winrules.c
--- a/src/winrules/winrules.c	2026-07-28 15:15:18.737967721 +0300
+++ b/src/winrules/winrules.c	2026-07-28 15:15:18.886490688 +0300
@@ -20,6 +20,7 @@
  */
 
 #include <compiz-core.h>
+#include <stdlib.h>
 
 #include <X11/Xatom.h>
 
diff --no-dereference -ruN a/src/workarounds/workarounds.c b/src/workarounds/workarounds.c
--- a/src/workarounds/workarounds.c	2026-07-28 15:15:18.736958400 +0300
+++ b/src/workarounds/workarounds.c	2026-07-28 15:15:18.890275717 +0300
@@ -21,6 +21,7 @@
  */
 
 #include <string.h>
+#include <stdlib.h>
 #include <limits.h>
 
 #include <compiz-core.h>
--- a/src/animation/wave.c	2026-07-28 15:15:18.738527077 +0300
+++ b/src/animation/wave.c	2026-07-28 15:15:18.855120754 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Wave  =========================
 
--- a/src/animation/dodge.c	2026-07-28 15:15:18.738501278 +0300
+++ b/src/animation/dodge.c	2026-07-28 15:15:18.818639828 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Dodge  =========================
 
--- a/src/animation/horizontalfold.c	2026-07-28 15:15:18.738469125 +0300
+++ b/src/animation/horizontalfold.c	2026-07-28 15:15:18.839491568 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Horizontal Folds  =========================
 
--- a/src/animation/magiclamp.c	2026-07-28 15:15:18.738444768 +0300
+++ b/src/animation/magiclamp.c	2026-07-28 15:15:18.843535096 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 void
 fxMagicLampInitGrid (CompWindow *w,
--- a/src/animation/dream.c	2026-07-28 15:15:18.738424634 +0300
+++ b/src/animation/dream.c	2026-07-28 15:15:18.822808557 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 Bool
 fxDreamAnimInit (CompWindow * w)
--- a/src/animation/curvedfold.c	2026-07-28 15:15:18.738391298 +0300
+++ b/src/animation/curvedfold.c	2026-07-28 15:15:18.814502247 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Curved Fold  =========================
 
--- a/src/animation/animation.c	2026-07-28 15:15:18.738296889 +0300
+++ b/src/animation/animation.c	2026-07-28 15:15:18.808530008 +0300
@@ -81,6 +81,7 @@
  */
 
 #include <GL/glu.h>
+#include <stdlib.h>
 #include "animation-internal.h"
 
 
--- a/src/animation/zoomside.c	2026-07-28 15:15:18.738236296 +0300
+++ b/src/animation/zoomside.c	2026-07-28 15:15:18.858814991 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Zoom and Sidekick  =========================
 
--- a/src/animation/options.c	2026-07-28 15:15:18.738193704 +0300
+++ b/src/animation/options.c	2026-07-28 15:15:18.847346125 +0300
@@ -26,6 +26,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 extern ExtensionPluginInfo animExtensionPluginInfo;
 extern AnimBaseFunctions animBaseFunctions;
--- a/src/animation/rollup.c	2026-07-28 15:15:18.738173982 +0300
+++ b/src/animation/rollup.c	2026-07-28 15:15:18.851125338 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Roll Up  =========================
 
--- a/src/animation/fade.c	2026-07-28 15:15:18.738155801 +0300
+++ b/src/animation/fade.c	2026-07-28 15:15:18.827523925 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Fade  =========================
 
--- a/src/animation/focusfade.c	2026-07-28 15:15:18.738136970 +0300
+++ b/src/animation/focusfade.c	2026-07-28 15:15:18.831862908 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Focus Fade  =========================
 
--- a/src/animation/glide.c	2026-07-28 15:15:18.738114818 +0300
+++ b/src/animation/glide.c	2026-07-28 15:15:18.835696592 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Glide  =========================
 
--- a/src/winrules/winrules.c	2026-07-28 15:15:18.737967721 +0300
+++ b/src/winrules/winrules.c	2026-07-28 15:15:18.886490688 +0300
@@ -20,6 +20,7 @@
  */
 
 #include <compiz-core.h>
+#include <stdlib.h>
 
 #include <X11/Xatom.h>
 
--- a/src/session/session.c	2026-07-28 15:15:18.737698139 +0300
+++ b/src/session/session.c	2026-07-28 15:15:18.878602086 +0300
@@ -25,6 +25,7 @@
 
 #define _GNU_SOURCE
 #include <X11/Xatom.h>
+#include <stdlib.h>
 #include <X11/SM/SM.h>
 
 #include <compiz-core.h>
--- a/src/mousepoll/mousepoll.c	2026-07-28 15:15:18.737002736 +0300
+++ b/src/mousepoll/mousepoll.c	2026-07-28 15:15:18.867250806 +0300
@@ -21,6 +21,7 @@
  */
 
 #include <compiz-core.h>
+#include <stdlib.h>
 
 #include "compiz-mousepoll.h"
 
--- a/src/workarounds/workarounds.c	2026-07-28 15:15:18.736958400 +0300
+++ b/src/workarounds/workarounds.c	2026-07-28 15:15:18.890275717 +0300
@@ -21,6 +21,7 @@
  */
 
 #include <string.h>
+#include <stdlib.h>
 #include <limits.h>
 
 #include <compiz-core.h>
--- a/src/mag/mag.c	2026-07-28 15:15:18.736543023 +0300
+++ b/src/mag/mag.c	2026-07-28 15:15:18.862830425 +0300
@@ -21,6 +21,7 @@
  */
 
 #include <math.h>
+#include <stdlib.h>
 #include <string.h>
 
 #include <compiz-core.h>
--- a/src/scaleaddon/scaleaddon.c	2026-07-28 15:15:18.736006437 +0300
+++ b/src/scaleaddon/scaleaddon.c	2026-07-28 15:15:18.874686476 +0300
@@ -25,6 +25,7 @@
  */
 
 #include <math.h>
+#include <stdlib.h>
 #include <string.h>
 #include <X11/Xatom.h>
 
--- a/src/opacify/opacify.c	2026-07-28 15:15:18.735797543 +0300
+++ b/src/opacify/opacify.c	2026-07-28 15:15:18.871026902 +0300
@@ -22,6 +22,7 @@
  */
 
 #include <compiz-core.h>
+#include <stdlib.h>
 #include "opacify_options.h"
 
 #define GET_OPACIFY_DISPLAY(d)                            \
--- a/src/vpswitch/vpswitch.c	2026-07-28 15:15:18.735274800 +0300
+++ b/src/vpswitch/vpswitch.c	2026-07-28 15:15:18.882827942 +0300
@@ -27,6 +27,7 @@
  */
 
 #include <compiz-core.h>
+#include <stdlib.h>
 #include <string.h>
 #include <X11/keysymdef.h>
 #include "vpswitch_options.h"
