--- a/src/animation/animation.c	2026-07-28 15:21:00.230823238 +0300
+++ b/src/animation/animation.c	2026-07-28 15:21:13.255138716 +0300
@@ -81,6 +81,7 @@
  */
 
 #include <GL/glu.h>
+#include <stdlib.h>
 #include "animation-internal.h"
 
 
--- a/src/animation/curvedfold.c	2026-07-28 15:21:00.230785206 +0300
+++ b/src/animation/curvedfold.c	2026-07-28 15:21:13.259501202 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Curved Fold  =========================
 
--- a/src/animation/dodge.c	2026-07-28 15:21:00.230657210 +0300
+++ b/src/animation/dodge.c	2026-07-28 15:21:13.263696233 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Dodge  =========================
 
--- a/src/animation/dream.c	2026-07-28 15:21:00.230760001 +0300
+++ b/src/animation/dream.c	2026-07-28 15:21:13.267385562 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 Bool
 fxDreamAnimInit (CompWindow * w)
--- a/src/animation/fade.c	2026-07-28 15:21:00.231061327 +0300
+++ b/src/animation/fade.c	2026-07-28 15:21:13.270988516 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Fade  =========================
 
--- a/src/animation/focusfade.c	2026-07-28 15:21:00.231076875 +0300
+++ b/src/animation/focusfade.c	2026-07-28 15:21:13.274494962 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Focus Fade  =========================
 
--- a/src/animation/glide.c	2026-07-28 15:21:00.231096765 +0300
+++ b/src/animation/glide.c	2026-07-28 15:21:13.278243603 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Glide  =========================
 
--- a/src/animation/horizontalfold.c	2026-07-28 15:21:00.230689951 +0300
+++ b/src/animation/horizontalfold.c	2026-07-28 15:21:13.281897694 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Horizontal Folds  =========================
 
--- a/src/animation/magiclamp.c	2026-07-28 15:21:00.230727862 +0300
+++ b/src/animation/magiclamp.c	2026-07-28 15:21:13.285394772 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 void
 fxMagicLampInitGrid (CompWindow *w,
--- a/src/animation/options.c	2026-07-28 15:21:00.231024976 +0300
+++ b/src/animation/options.c	2026-07-28 15:21:13.288928860 +0300
@@ -26,6 +26,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 extern ExtensionPluginInfo animExtensionPluginInfo;
 extern AnimBaseFunctions animBaseFunctions;
--- a/src/animation/rollup.c	2026-07-28 15:21:00.231044864 +0300
+++ b/src/animation/rollup.c	2026-07-28 15:21:13.292331113 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Roll Up  =========================
 
--- a/src/animation/wave.c	2026-07-28 15:21:00.230627924 +0300
+++ b/src/animation/wave.c	2026-07-28 15:21:13.295639613 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Wave  =========================
 
--- a/src/animation/zoomside.c	2026-07-28 15:21:00.230987537 +0300
+++ b/src/animation/zoomside.c	2026-07-28 15:21:13.299063072 +0300
@@ -35,6 +35,7 @@
  */
 
 #include "animation-internal.h"
+#include <stdlib.h>
 
 // =====================  Effect: Zoom and Sidekick  =========================
 
--- a/src/mag/mag.c	2026-07-28 15:21:00.232555643 +0300
+++ b/src/mag/mag.c	2026-07-28 15:21:13.302539128 +0300
@@ -21,6 +21,7 @@
  */
 
 #include <math.h>
+#include <stdlib.h>
 #include <string.h>
 
 #include <compiz-core.h>
--- a/src/mousepoll/mousepoll.c	2026-07-28 15:21:00.232135719 +0300
+++ b/src/mousepoll/mousepoll.c	2026-07-28 15:21:13.306231656 +0300
@@ -21,6 +21,7 @@
  */
 
 #include <compiz-core.h>
+#include <stdlib.h>
 
 #include "compiz-mousepoll.h"
 
--- a/src/opacify/opacify.c	2026-07-28 15:21:00.233251733 +0300
+++ b/src/opacify/opacify.c	2026-07-28 15:21:13.309690353 +0300
@@ -22,6 +22,7 @@
  */
 
 #include <compiz-core.h>
+#include <stdlib.h>
 #include "opacify_options.h"
 
 #define GET_OPACIFY_DISPLAY(d)                            \
--- a/src/scaleaddon/scaleaddon.c	2026-07-28 15:21:00.233062882 +0300
+++ b/src/scaleaddon/scaleaddon.c	2026-07-28 15:21:13.313371626 +0300
@@ -25,6 +25,7 @@
  */
 
 #include <math.h>
+#include <stdlib.h>
 #include <string.h>
 #include <X11/Xatom.h>
 
--- a/src/session/session.c	2026-07-28 15:21:00.231482435 +0300
+++ b/src/session/session.c	2026-07-28 15:21:13.317574061 +0300
@@ -25,6 +25,7 @@
 
 #define _GNU_SOURCE
 #include <X11/Xatom.h>
+#include <stdlib.h>
 #include <X11/SM/SM.h>
 
 #include <compiz-core.h>
--- a/src/vpswitch/vpswitch.c	2026-07-28 15:21:00.233702056 +0300
+++ b/src/vpswitch/vpswitch.c	2026-07-28 15:21:13.321666626 +0300
@@ -27,6 +27,7 @@
  */
 
 #include <compiz-core.h>
+#include <stdlib.h>
 #include <string.h>
 #include <X11/keysymdef.h>
 #include "vpswitch_options.h"
--- a/src/winrules/winrules.c	2026-07-28 15:21:00.231223304 +0300
+++ b/src/winrules/winrules.c	2026-07-28 15:21:13.325319359 +0300
@@ -20,6 +20,7 @@
  */
 
 #include <compiz-core.h>
+#include <stdlib.h>
 
 #include <X11/Xatom.h>
 
--- a/src/workarounds/workarounds.c	2026-07-28 15:21:00.232168385 +0300
+++ b/src/workarounds/workarounds.c	2026-07-28 15:21:13.328856410 +0300
@@ -21,6 +21,7 @@
  */
 
 #include <string.h>
+#include <stdlib.h>
 #include <limits.h>
 
 #include <compiz-core.h>
