--- a/src/main.c
+++ b/src/main.c
@@ -271,8 +271,6 @@
     i += compReadXmlChunk ("</screen></core></compiz>", &offset, buffer + i,
 			   length - i);
 
-    if (i > 0 && !offset && length > i)
-	buffer[i++] = '\0';
 
     ctx->offset += i;
 
--- a/src/metadata.c
+++ b/src/metadata.c
@@ -286,8 +286,6 @@
     i += compReadXmlChunk ("</plugin></compiz>", &offset, buffer + i,
 			   length - i);
 
-    if (i > 0 && !offset && length > i)
-	buffer[i++] = '\0';
 
     ctx->offset += i;
 
