--- asus_oled.c
+++ asus_oled.c
@@ -574,10 +574,10 @@
 }
 
 static DEVICE_ATTR_RO(version);
-static DEVICE_ATTR(asus_oled_enabled, S_IWUGO | S_IRUGO, get_enabled, set_enabled);
-static DEVICE_ATTR(asus_oled_picture, S_IWUGO , NULL, set_picture);
-static DEVICE_ATTR(enabled, S_IWUGO | S_IRUGO, class_get_enabled, class_set_enabled);
-static DEVICE_ATTR(picture, S_IWUGO, NULL, class_set_picture);
+static DEVICE_ATTR(asus_oled_enabled, 0644, get_enabled, set_enabled);
+static DEVICE_ATTR(asus_oled_picture, 0200, NULL, set_picture);
+static DEVICE_ATTR(enabled, 0644, class_get_enabled, class_set_enabled);
+static DEVICE_ATTR(picture, 0200, NULL, class_set_picture);
 
 static int asus_oled_probe(struct usb_interface *interface,
 	const struct usb_device_id *id)
@@ -725,7 +725,7 @@
 
 static ssize_t version_show(struct class *class, struct class_attribute *attr, char *buf)
 {
-	return snprintf(buf, PAGE_SIZE, "%s\n", ASUS_OLED_VERSION);
+	return sysfs_emit(buf, "%s\n", ASUS_OLED_VERSION);
 }
 
 static CLASS_ATTR_RO(version);
@@ -735,7 +735,7 @@
 	int retval;
 	struct usb_interface *interface;
 
-	oled_class = class_create(THIS_MODULE, ASUS_OLED_UNDERSCORE_NAME);
+	oled_class = class_create(ASUS_OLED_UNDERSCORE_NAME);
 	if (IS_ERR(oled_class)) {
 		pr_err("Error creating " ASUS_OLED_UNDERSCORE_NAME " class\n");
 		return PTR_ERR(oled_class);
@@ -751,7 +751,7 @@
 	retval = usb_register(&asus_oled_driver);
 	if (retval) {
 		pr_err("usb_register failed. Error number %d\n", retval);
-		class_remove_file(oled_class, &class_attr_version);
+		class_destroy(oled_class);
 	}
 	return retval;
 }
@@ -761,7 +761,7 @@
 {
 	struct usb_interface *interface;
 
-	class_remove_file(oled_class, &class_attr_version);
+	/* class_attr_version removed with class_destroy */
 	class_destroy(oled_class);
 	usb_deregister(&asus_oled_driver);
 }
