--- a/src/wallpaper/wallpaper.c
+++ b/src/wallpaper/wallpaper.c
@@ -35,6 +35,8 @@
 
 #include "wallpaper_options.h"
 
+#include <stdlib.h>
+
 typedef struct _WallpaperBackground
 {
     CompSize size;
--- a/src/addhelper/addhelper.c
+++ b/src/addhelper/addhelper.c
@@ -26,6 +26,8 @@
 
 #include "addhelper_options.h"
 
+#include <stdlib.h>
+
 typedef struct _AddHelperDisplay
 {
     int		    screenPrivateIndex;
